src/apply.rs

```diff
@@ -0,0 +1,36 @@
+use divide;
+
+///Parallelizes `operation` over given `data`. 
+///
+///The data are divided into chunks based on the number of processing cores 
+///available to the rust process according to `std::os::num_cpus`. 
+///
+///# Panics
+///
+///Panics if any of the underlying threads panics
+///
+///# Example
+///
+/// ```
+/// extern crate parallel;
+/// 
+/// # fn main() {
+/// let mut v = (1..10).collect::<Vec<usize>>();
+/// # let w = v.iter().map(|x| x + 1).collect::<Vec<usize>>();
+/// parallel::apply(v.as_mut_slice(), |x| {
+/// 	*x = *x + 1;
+///     //.. 
+/// });
+/// # assert_eq!(v, w);
+/// # }
+/// ```
+pub fn apply<T, F>(data: &mut [T], operation: F) where
+	T: Send,
+	F: Fn(&mut T) + Sync,
+{
+	divide::divide(data, None, |data, _|{
```

Comment:
I think you can directly use `::divide(..)` here, instead of importing the `divide` module.